doc: update migration guide and release notes for PRs #71827 and #72243 (#73508)
zephyrproject-rtos/zephyr: doc/releases/release-notes-3.7.rst

  * Removed IPSP support. ``CONFIG_NET_L2_BT`` does not exist anymore.

* TCP:

>   * ISN generation use SHA256 instead of MD5. Moreover it relies on PSA APIs

Reviewer: ```suggestion
  * ISN generation uses SHA256 instead of MD5. Moreover it relies on PSA APIs
```

>   * ISN generation uses SHA256 instead of MD5. Moreover it relies on PSA APIs

Reviewer 2: ```suggestion
  * ISN generation now uses SHA256 instead of MD5. Moreover it now relies on PSA APIs
```
to express the change (could be done differently)

>   * ISN generation uses now SHA256 instead of MD5. Moreover it now relies on PSA APIs

Reviewer 2: ```suggestion
  * ISN generation now uses SHA256 instead of MD5. Moreover it now relies on PSA APIs
```

>   * ISN generation now uses SHA256 instead of MD5. Moreover it now relies on PSA APIs

Reviewer 3: ```suggestion
  * ISN generation now uses SHA-256 instead of MD5. Moreover it now relies on PSA APIs
```

SHA-256 is used elsewhere and this is often times added in the spec-definition of the algorithm (more searchable)

    instead of legacy Mbed TLS functions for hash computation.

> 

Reviewer 2: Wondering whether this should rather be in the migration guide as the previous Kconfig option behavior (which used to `select MBEDTLS` and others) is changed?

Author: Actually the `migration-guide` has an entry for this change as well [here](https://github.com/zephyrproject-rtos/zephyr/pull/73508/files#diff-f80404b1ec33e5720fc31f6c388acee451f54148e7fff15cd55ed94b3546f1cfR554). Based on your comment, I think I can extend it to mention the Kconfig change though

  * Mbed TLS was updated to 3.6.0. Release notes can be found at:
    https://github.com/Mbed-TLS/mbedtls/releases/tag/v3.6.0
  * When any PSA crypto provider is available in the system, desired PSA features

>     must be explicitly selected through `CONFIG_PSA_WANT_xxx` symbols. Moreover

Reviewer: ```suggestion
    must be explicitly selected through ``CONFIG_PSA_WANT_xxx`` symbols. Moreover
```

    TLS/X509/PK/MD modules will make use of PSA APIs instead of legacy Mbed TLS

>     ones.

Reviewer 2: > Moreover TLS/X509/PK/MD modules will make use of PSA APIs instead of legacy Mbed TLS ones.

This was already the case at least in the `CONFIG_MBEDTLS_PSA_CRYPTO_C` case: https://github.com/zephyrproject-rtos/zephyr/pull/71827/files#diff-35af03c2e234822e54b1a5ef8918c465ca47f246d1dc732740e125e6a2935732R475-R479

  * Mbed TLS was updated to 3.6.0. Release notes can be found at:
    https://github.com/Mbed-TLS/mbedtls/releases/tag/v3.6.0

>   * When any PSA crypto provider is available in the system, desired PSA features

Reviewer 2: ```suggestion
  * When any PSA crypto provider is available in the system (:kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_CLIENT` is enabled), desired PSA features
```

>     must be explicitly selected through ``CONFIG_PSA_WANT_xxx`` symbols.

Reviewer 2: ```suggestion
    must now be explicitly selected through ``CONFIG_PSA_WANT_xxx`` symbols.
```

  * Mbed TLS was updated to 3.6.0. Release notes can be found at:
    https://github.com/Mbed-TLS/mbedtls/releases/tag/v3.6.0
  * When any PSA crypto provider is available in the system
    (:kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_CLIENT` is enabled), desired PSA features
    must now be explicitly selected through ``CONFIG_PSA_WANT_xxx`` symbols.
  * Choice symbols :kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_LEGACY_RNG` and
    :kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG` were added in order
    to allow the user to specify how Mbed TLS PSA crypto core should generate random numbers.

>     The former option, which is the default, relies on legacy entropy and CTR/HMAC DRBG modules,

Reviewer 3: ```suggestion
    The former option, which is the default, relies on legacy entropy and CTR_DRBG/HMAC_DRBG modules,
```

The underscore is present in the NIST spec of these and these are better searchable terms (don't add additional abbreviations that aren't searchable)
